lib/perl/Genome/Model/Build/Command/SymlinkResults.pm

```diff
@@ -0,0 +1,50 @@
+package Genome::Model::Build::Command::SymlinkResults;
+
+use strict;
+use warnings;
+
+use Genome;
+
+class Genome::Model::Build::Command::SymlinkResults {
+    is => ['Command::V2'],
+    has => [
+        build => {
+            is => 'Genome::Model::Build',
+            shell_args_position => 1,
+            doc => "The builds that has results you'd like to create symlinks to",
+        },
+        destination => {
+            is => 'DirectoryPath',
+            shell_args_position => 2,
+            doc => 'The desired location under which to make a directory of symlink(s)',
+        }
+    ],
+    doc => 'Creates symlinks to the results of a build.',
+};
+
+sub help_detail {
+    my $self = shift;
+    return <<EOP;
+Creates symlinks to the results of a build.  The <destination> option should
+be an existing directory.
+EOP
+}
+
+sub execute {
+    my $self = shift;
+    my $build = $self->build;
+    my $destination = $self->destination;
+
+    Genome::Sys->validate_directory_for_read_write_access($destination);
+
+    unless($build->can('symlink_results')) {
+        $self->fatal_message('Symlinking results is unavailable for builds of type "%s"', $build->type_name);
+    }
+
```

Comment:
I am not sure what facilities this object model has for traits/roles/flavors, but it is generally better to validate a class has a particular interface than to trust duck typing (does vs can).  $dog->bark vs $tree->bark and all that.  Would that be possible here?

An alternative to an interface would be to assume that all build types support a symlink_results method, and the default implementation in the base class could be to either serve the fatal message or NOP.